vimeo/VimeoNetworking
# [WIP] VIM-XXXX: Introduces SessionManaging protocol for dependency abstraction

#### Ticket

[VIM-XXXX](https://vimean.atlassian.net/browse/VIM-XXXX)

#### Pull Request Checklist


#### Issue Summary

This PR introduces several protocol declarations with the main goal of making it easier to abstract and eventually remove our dependency on `AFNetworking`.

#### Implementation Summary

The key changes to pay attention to can be seen in `VimeoClient`, where we replaced the dependency on `VimeoSessionManager` by a composition of 2 protocols: `SessionManaging` and `AuthenticationListenerDelegate`.

Also some of the interactions with the `VimeoSessionManager` that previously occurred inside `VimeoClient` have been moved to the session manager subclass itself. This greatly simplified the protocol requirements for `SessionManaging` to a single `request` method that takes an `EndPointType` and a callback closure.

In a future PR this protocol will be extended to also allow for `Decodable` types to be serialized.

#### How to Test

This PR doesn't introduce any real functional changes so simply ensure that you can build & run, and that all tests pass. Also please take it for a spin in the app by checking out [this branch](feature/VIM-XXXX_AFNetworkingDeprecation).
